ta/pkcs11/src/processing_symm.c

```diff
@@ -1019,8 +1019,10 @@ enum pkcs11_rc derive_key_by_symm_enc(struct pkcs11_session *session,
 		res = TEE_CipherDoFinal(proc->tee_op_handle, in_buf, in_size,
 					*out_buf, out_size);
 		rc = tee2pkcs_error(res);
-		if (rc)
+		if (rc) {
 			TEE_Free(*out_buf);
+			*out_buf = NULL;
```

Comment:
`*out_buf` output value is not expected to be freed by caller since the function failed.
But I see that implementation of caller does free the buffer! Nice catch.

A better fix, IMO, would be to not load a reference in `*out_buf` unless the function succeeds.
Yet the fix proposed here does the job.

Reply:
I agree that relying on `TEE_Free(NULL)` beind valid is not the prettiest solution. I'll change the fix to be more invasive, as you've suggested.